### DISCUSSION TITLE
Why are changed files not being marked as unchanged in git?

### DISCUSSION THREAD

**Discussion by poperigby** [twpayne/chezmoi#1309] [Q&A] [answered]
I'll change a file in the target directory (e.g. `~/.config/nvim/init.lua`), but it doesn't show anything when I run `chezmoi git status`. Why is this?

**Comment by twpayne** [comment, marked as answer]
`chezmoi git status` runs `git status` in the source directory. Did you mean to run `chezmoi status` instead?

**Reply by poperigby** [reply]
I'm probably misunderstanding how this works. Does Chezmoi not track the changes in its git repository when the target is changed?

**Reply by nandalopes** [reply]
Nope. You must merge/re-add the files you've changed. But chezmoi don't overwrite existing and newer files without prompt either.

**Reply by poperigby** [reply]
Is that what `chezmoi re-add` is for?

**Reply by twpayne** [reply]
What @nandalopes said.

For a sequence diagram, see [this video](https://mirrors.dotsrc.org/fosdem/2021/L.lightningtalks/chezmoi.webm) at 5m44s.

**Reply by twpayne** [reply]
> Is that what `chezmoi re-add` is for?

`chezmoi re-add` is one option, but not necessary. For options see [this FAQ entry](https://github.com/twpayne/chezmoi/blob/master/docs/FAQ.md#how-do-i-edit-my-dotfiles-with-chezmoi).

**Reply by poperigby** [reply]
Thank you. I think I just need to try wrapping my head around the differences between this and yadm.

**Reply by poperigby** [reply]
Well, actually one more thing. How do I add deleted files?

**Reply by twpayne** [reply]
> Thank you. I think I just need to try wrapping my head around the differences between this and yadm.

One thing that might help is to know that chezmoi _generates_ your dotfiles in the target directory from the contents of the source directory. Only the source directory is managed with git.

> Well, actually one more thing. How do I add deleted files?

Do you mean how do you make chezmoi delete files? If so, look at [`.chezmoiremove`](https://github.com/twpayne/chezmoi/blob/master/docs/HOWTO.md#ensure-that-a-target-is-removed).

**Reply by poperigby** [reply]
> One thing that might help is to know that chezmoi generates your dotfiles in the target directory from the contents of the source directory. Only the source directory is managed with git.

Okay, that makes sense.

> Do you mean how do you make chezmoi delete files? If so, look at .chezmoiremove.

I just wanted to tell chezmoi that I deleted a target file, but I figured out that `chezmoi forget` does that.
